src/association/mod.rs

```diff
@@ -238,21 +240,14 @@ impl<P> fmt::Display for Association<P> {
 
 impl<P: HardSphereProperties> Association<P> {
     pub fn assoc_site_frac_ab<D: DualNum<f64>>(deltarho: D, na: f64, nb: f64) -> D {
-        if deltarho.re() > f64::EPSILON.sqrt() {
-            (((deltarho * (na - nb) + 1.0).powi(2) + deltarho * nb * 4.0).sqrt()
-                - (deltarho * (nb - na) + 1.0))
-                / (deltarho * na * 2.0)
-        } else {
-            D::one() + deltarho * nb * (deltarho * (nb + na) - 1.0)
-        }
+        (((deltarho * (na - nb) + 1.0).powi(2) + deltarho * nb * 4.0).sqrt()
```

Comment:
with this slightly rewritten expression, there is no 0/0 issue anymore when the association strength or density go towards 0.